language/functional_tests/tests/testsuite/natives/vector/vector_resource_not_destroyed_at_return.mvir

```diff
@@ -0,0 +1,13 @@
+module M {
+    import 0x0.Vector;
+
+    resource R {}
+
+    f() {
+        let v: Vector.T<Self.R>;
+        v = Vector.empty<Self.R>();
+        return;
+    }
+}
+
+// check: RET_UNSAFE_TO_DESTROY_ERROR
```

Comment:
Might be good to have a variant of this test where we let a `Vector.T<u64>` go out of scope without calling `destroy_empty`?